extensions/amp-story/0.1/amp-story.js

```diff
@@ -478,6 +478,16 @@ export class AmpStory extends AMP.BaseElement {
       this.onKeyDown_(e);
     }, true);
 
+    // keep track of page-id in browser history API
+    this.storeService_.subscribe(StateProperty.CURRENT_PAGE_ID, pageId => {
+      const history = this.win.history;
```

Comment:
I think it'll be hard to justify the design of why we handle the browser history in the ``initializeListeners`` of the main class to future developers haha.
What do you think of sticking to the pattern that's right above this subscribe method call:

````
this.storeService_.subscribe(StateProperty.CURRENT_PAGE_ID, pageId => {
  this.onCurrentPageIdUpdate_(pageId);
});
````

I do agree with you that this feature doesn't need to do anything fancy. But how about, from this handler, we use a very simple API like:

````
onCurrentPageIdUpdate_() {
  this.setHistoryState({...});
}

setHistoryState(state) {}

getHistoryState() {
  return state || {};
}
````

It's very likely that we will need to do more things ``onCurrentPageIdUpdate``, and keeps the methods doing one thing and doing them well.

Reply:
moved to separate methods